src/main.cpp

```diff
@@ -2,62 +2,358 @@
-    auto startTime = std::chrono::high_resolution_clock::now();
-    for (int i = 0; i < iterations; i++) {
-        pipeline.Go(input);
-    }
-    auto endTime = std::chrono::high_resolution_clock::now();
-    auto totalTime = std::chrono::duration<double, std::milli>(endTime - startTime);
-    std::cout << "total_ms " << totalTime.count() << std::endl;
-}
+// DO NOT DELETE
+// static void PipelineBenchmark() {
+//     PipelineInputList input = PromptPipelineInput();
+//     Pipeline pipeline = PromptPipeline();
+//     int iterations = Prompt<int>("Times to run the pipeline");
+//     std::cerr << "Benchmarking..." << std::endl;
+
+//     // TODO: we can do better than this :| maybe include mean time, 99% time, or allow a vector of
+//     // input and determine which one took the longest
+//     auto startTime = std::chrono::high_resolution_clock::now();
+//     for (int i = 0; i < iterations; i++) {
+//         pipeline.Go(input);
+//     }
+//     auto endTime = std::chrono::high_resolution_clock::now();
+//     auto totalTime = std::chrono::duration<double, std::milli>(endTime - startTime);
+//     std::cout << "total_ms " << totalTime.count() << std::endl;
+// }
 
 }
 
 int main(int argc, char **argv) {
-    lost::RegisterCliArgs(argc, argv);
-    std::cerr << "LOST: Open-source Star Tracker" << std::endl;
-    lost::InteractiveChoice<void (*)()> mainChoices;
-    mainChoices.Register("pipeline", "Run a pipeline", &lost::PipelineRun);
-    mainChoices.Register("benchmark", "Benchmark a pipeline", &lost::PipelineBenchmark);
-    mainChoices.Register("build_database", "Build database from catalog", &lost::DatabaseBuild);
-    (*mainChoices.Prompt("Choose action"))();
-}
+
+    if (argc == 1) {
+        std::cout << "Usage: ./lost database or ./lost pipeline" << std::endl << "Use --help flag on those commands for further help" << std::endl; 
+        return 0;
+    } 
+
+    std::string command (argv[1]);
+    if (command == "database") {
+
+        enum DatabaseEnum {mag, stars, kvector, kvectorMinDistance, kvectorMaxDistance, 
+            kvectorDistanceBins, help, output};
+
+        static struct option long_options[] =
+        {
+            {"mag",     required_argument, 0, mag},
+            {"stars",   required_argument, 0, stars},
+            {"kvector", no_argument,       0, kvector},
+            {"kvector-min-distance", required_argument, 0, kvectorMinDistance},
+            {"kvector-max-distance",  required_argument, 0, kvectorMaxDistance},
+            {"kvector-distance-bins",  required_argument, 0, kvectorDistanceBins},
+            {"help",  no_argument, 0, help},
+            {"output", required_argument, 0, output},
+            {0, 0, 0, 0}
+        };
+
+        lost::DatabaseOptions databaseOptions;
+        int index;
+        int option;
+
+        while (optind < argc && argc != 2) {
+            if (optind < 2) optind++;
+
+            if ((option = getopt_long(argc, argv, "", long_options, &index)) != -1) {
```

Comment:
This right here is why you get hangs when  -- when `option == -1`: The while loop keeps going indefinitely, and getopt_long keeps returning -1. When it returns -1, you need to break the loop and print an error.